src/ResourceManagement/HDInsight/HDInsight.Tests/UnitTests/ClusterCreateTests.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Microsoft.Azure.Management.HDInsight;
+using System.Collections.Generic;
+using Xunit;
+using Microsoft.Azure.Management.HDInsight.Models;
+using Newtonsoft.Json;
+
+namespace HDInsight.Tests.UnitTests
+{
+    public class ClusterCreateTests
+    {
+        public ClusterCreateTests()
+        {
+        }
+
+        [Fact]
+        public void TestCreateDefaultFsAzureBlobClusterUsingClusterParameters()
+        {
+            string clusterName = "cluster";
+            string storageContainerName = "storageContainer";
+
+            var clusterCreateParams = new ClusterCreateParameters { 
+                ClusterSizeInNodes = 3,
+                ClusterType = "Hadoop",
+                WorkerNodeSize = "Large",
+                DefaultStorageInfo = new AzureStorageInfo("storageAccountName", "storageAccountKey", storageContainerName),
+                OSType = OSType.Linux,
+                UserName = "HttpUser",
+                Password = "HttpPassword",
+                Location = "East US",
+                SshUserName = "SshUser",
+                SshPassword = "SshPassword",
+                Version = "3.2"
+            };
+
+            ClusterOperations op = new ClusterOperations(new HDInsightManagementClient());
+            var extendedParams = op.GetExtendedClusterCreateParameters(clusterName, clusterCreateParams);
+            Assert.True(ContainerMatchesNameProvided(extendedParams, storageContainerName), "Container name does not match the one specified");
+        }
+
+        [Fact]
+        public void TestCreateDefaultFsAzureBlobClusterContainerNotSpecified()
+        {
+            string clusterName = "cluster";
+
+            var clusterCreateParams = new ClusterCreateParameters
+            {
+                ClusterSizeInNodes = 3,
+                ClusterType = "Hadoop",
+                WorkerNodeSize = "Large",
+                DefaultStorageInfo = new AzureStorageInfo("storageAccountName", "storageAccountKey"),
+                OSType = OSType.Linux,
+                UserName = "HttpUser",
+                Password = "HttpPassword",
+                Location = "East US",
+                SshUserName = "SshUser",
+                SshPassword = "SshPassword",
+                Version = "3.2"
+            };
+
+            ClusterOperations op = new ClusterOperations(new HDInsightManagementClient());
+            var extendedParams = op.GetExtendedClusterCreateParameters(clusterName, clusterCreateParams);
+            Assert.True(ContainerMatchesNameProvided(extendedParams, clusterName), "Container name does not match the cluster name");
+        }
+
+        private bool ContainerMatchesNameProvided(ClusterCreateParametersExtended x, string name)
```

Comment:
If you really need two different errors, then please pass that in, though in my opinion the regular assert failure is fine.

Either way, I'd rather have it be self sufficient rather than forcing the user to assert when that is the only purpose this serves.